openmls/src/schedule/message_secrets.rs

```diff
@@ -0,0 +1,89 @@
+//! This module defines the [`MessageSecrets`] struct that can be used for message decryption & verification
+
+use super::*;
+#[cfg(test)]
+use crate::tree::index::LeafIndex;
+/// Combined message secrets that need to be stored for later decryption/verification
+#[derive(Debug, Serialize, Deserialize)]
+pub struct MessageSecrets {
+    pub(crate) sender_data_secret: SenderDataSecret,
+    pub(crate) membership_key: MembershipKey,
+    pub(crate) confirmation_key: ConfirmationKey,
+    pub(crate) serialized_context: Vec<u8>,
+    pub(crate) secret_tree: SecretTree,
+}
+
+// Public functions
+impl MessageSecrets {
+    /// Get a reference to the message secrets's sender data secret.
+    pub fn sender_data_secret(&self) -> &SenderDataSecret {
```

Comment:
It's a little odd to have getter for all the fields but no way of setting them, i.e. for creating you rely on the `pub(crate)` visibility of the fields but not to get the values. Maybe make the fields private and add a constructor?

Reply:
The `secret_tree` can lose the `pub(crate)` as well.

Reply:
oops, fixed